platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DefaultConfigurationCache.kt

```diff
@@ -410,79 +413,105 @@ class DefaultConfigurationCache internal constructor(
 
     private
     fun saveModel(model: Any) {
-        saveToCache(
-            stateType = StateType.Model
-        ) { stateFile -> cacheIO.writeModelTo(model, stateFile) }
-    }
+        cacheEntryRequiresCommit = true
 
-    private
-    fun saveWorkGraph() {
-        saveToCache(stateType = StateType.Work) { stateFile ->
-            writeConfigurationCacheState(stateFile)
+        // TODO:configuration-cache should this flag affect model storing behavior?
+        if (startParameter.isIgnoreInputsInTaskGraphSerialization) {
+            InstrumentedInputs.discardListener()
         }
+
+        buildOperationRunner.withModelStoreOperation {
```

Comment:
This is the main change: a separate build operation is used to store the model. Same for the load operation.